### ISSUE TITLE
 Why use  weights = weights.detach() on line 137 of the file common.py?

### ISSUE THREAD

**Issue by xiruiYan** [yifita/deep_cage#16]
Hi,  thanks for your excellent results on this paper and the published code. I try to do some experiment with your code. I successfully train the network, but I have a little doubt.

On line 137 of the file common.py , you used: **weights = weights.detach().**
And in file losses.py, you used: **self.loss["WREG"]+=self.mvc_reg_loss(all_outputs["weight"])*self.opt.mvc_weight**
While **all_outputs[“weight”].requires_grad == False** here, I am confused about whether loss WREG affects the final result. 
From my experiment , I find that no matter how big the value of **mvc_weight** is, the result has not changed. 

Looking forward to your reply, thank you.

**Comment by yifita** [comment]
Hi Xirui,

Well spotted, thank you! After checking the dev repo, I found this line in common.py was indeed added later, out of context. I'm going to comment it out.

Thank you very much for point it out!!
